src/Antiforgery/src/Internal/DefaultAntiforgeryTokenStore.cs

```diff
@@ -39,16 +39,19 @@ public async Task<AntiforgeryTokenSet> GetRequestTokensAsync(HttpContext httpCon
 
         var cookieToken = httpContext.Request.Cookies[_options.Cookie.Name!];
 
+        // endpoint could override global request token source option
+        var tokenSource = httpContext.GetEndpoint()?.Metadata.GetMetadata<RequireAntiforgeryTokenAttribute>()?.RequestTokenSourceValue ?? _options.RequestTokenSource;
```

Comment:
What if you have something like this?

```csharp
[RequireAntiforgeryToken(RequestTokenSource = AntiforgeryRequestTokenSource.Header)]
public class MyController
{
    [HttpPost]
    [RequireAntiforgeryToken]
    public IActionResult Post() { ... }
}
```

Based on this line, it looks like it will resolve the most specific `RequireAntiforgeryTokenAttribute` which would be the one on the `Post()` method and see that the `RequestTokenSourceValue == null` and fall back to `_options.RequestTokenSource` which is `HeaderOrFormBody` when it ought to be using `Header` as specified by the attribute on `MyController`.

Reply:
I will look into this scenario